src/main/java/com/javadeobfuscator/deobfuscator/transformers/stringer/HideAccessObfuscationTransformer.java

```diff
@@ -228,6 +372,52 @@ public boolean canCheckEquality(JavaValue first, JavaValue second, Context conte
 
                         if (type1.getClassName().equals(type2.getClassName())) {
                             methodNode.instructions.remove(checkcast);
+                        }else
+                        {
+	                        String class2pre = type2.getClassName().replace(".", "/");
```

Comment:
This code can be cleaned up a lot. For example, `Type#getInternalName` instead of `type.getClassName().replace(".", "/")`, `Type#getSort` instead of `type.getClassName().endsWith("[]")`, etc.